tests/drivers/gpio/gpio_basic_api/testcase.yaml

```diff
@@ -53,3 +65,30 @@ tests:
     filter: dt_compat_enabled("test-gpio-basic-api") and dt_compat_enabled("arduino-header-r3")
     platform_allow:
       - mimxrt595_evk/mimxrt595s/cm33
+      - mimxrt1020_evk
+      - mimxrt1040_evk
+      - mimxrt1060_evk
+      - mimxrt1060_evkb
+      - lpcxpresso55s69/lpc55s69/cpu0
+      - lpcxpresso55s36
+      - nucleo_wb09ke
+      - nucleo_wb05kz
+      - nucleo_wba55cg
+      - nucleo_l031k6
```

Comment:
From what I understood, only wb0 boards required a specific overlay. @djiatsaf-st can you clarify the status ?

Reply:
this is not a must, as it also helps to demo how to customize the test settings. agree? @erwango

Reply:
> From what I understood, only wb0 boards required a specific overlay. @djiatsaf-st can you clarify the status ?

There will be no problem for nucleo_l031k6.
For nucleo_wba55cg, the user manual specifies that we need to change state of solder bridges to use pin D8 or D9. 
So it will be better to leave as excluded.

Reply:
@hakehuang Please make `nucleo_l031k6` use st_arduino.overlay  and we'll be good to go

Reply:
done